src/TitleDetails/AddPieceModal/AddPieceModal.js

```diff
@@ -150,15 +146,13 @@ const AddPieceModal = ({
         </Row>
         <Row>
           <Col xs>
-            <FieldSelectFinal
-              dataOptions={locations}
-              disabled
-              fullWidth
-              label={<FormattedMessage id="ui-receiving.piece.location" />}
+            <FieldLocationFinal
+              id={locationId}
```

Comment:
bad naming, it's not clear why do you set `id` prop, that usually id in DOM as location id.